attestation-service/attestation-service/src/lib.rs

```diff
@@ -72,44 +103,50 @@ impl AttestationService {
             .map_err(|e| anyhow!("Cannot Set Policy: {:?}", e))
     }
 
-    fn accumulate_hash(materials: &[Vec<u8>]) -> Option<Vec<u8>> {
-        if materials.is_empty() {
-            return None;
-        }
-        let mut hasher = Sha384::new();
-        materials.iter().for_each(|m| hasher.update(m));
-        Some(hasher.finalize().to_vec())
-    }
-
     /// Evaluate Attestation Evidence.
-    /// Issue an attestation results token which contain TCB status and TEE public key.
-    pub async fn evaluate(&self, tee: Tee, nonce: &str, attestation: &str) -> Result<String> {
-        let attestation = serde_json::from_str::<Attestation>(attestation)
-            .context("Failed to deserialize Attestation")?;
+    /// Issue an attestation results token which contain TCB status and TEE public key. Input parameters:
+    /// - `evidence`: TEE evidence bytes. This might not be the raw hardware evidence bytes. Definitions
+    /// are in `verifier` crate.
+    /// - `tee`: concrete TEE type
+    /// - `runtime_data`: plaintexts of the runtime_data. These data will be hased using the specified algorithm
+    /// to calculate a digest. Then the digest will be used to match the `report_data` field of the evidence.
+    /// - `init_data`: plaintexts of the init_data. These data will be hased using the specified algorithm
+    /// to calculate a initdata hash. Then the initdata hash will be used to match the `init_data_hash` field of the
+    /// evidence.
```

Comment:
we need to have a better way to differentiate between "no runtime data" and "empty runtime data". and same with initdata. Hash of "" is not all zeros should be valid and is different than all zeros.

Reply:
i think so